scripts/apatch.sh

```diff
@@ -1,75 +1,88 @@
 
-noapply=1
-isreject=0
-if [[ $1 == "--noapplied" ]]; then
-	noapply=1
-	shift
+# We define gitcmd and some helper functions in here.
+. "$basedir"/scripts/functions.sh
+
+noapply=true
+isreject=false
+if [ "${1:-}" = "--noapplied" ]; then
+    noapply=true
+    shift
 fi
 
-if [ ! -z "$1" ]; then
-	file="$1"
-elif [ -z "$1" ] && [ -f .git/rebase-apply/patch ]; then
-	file=".git/rebase-apply/patch"
-	noapply=1
-	isreject=1
+if [ -n "${1:-}" ]; then
+    file="$1"
+elif [ -f .git/rebase-apply/patch ]; then
+    file=".git/rebase-apply/patch"
+    noapply=true
+    isreject=true
 else
-	echo "Please specify a file"
-	exit 1
-fi
-applied=$(echo $file | sed 's/.patch$/-applied\.patch/g')
-if [ "$1" == "--reset" ]; then
-	$gitcmd am --abort
-	$gitcmd reset --hard
-	$gitcmd clean -f
-	exit 0
+    echo "Please specify a file"
+    exit 1
 fi
 
+applied=$(echo "$file" | sed 's/.patch$/-applied\.patch/g')
+if [ "${1:-}" = "--reset" ]; then
+    $gitcmd am --abort
+    $gitcmd reset --hard
+    $gitcmd clean -f
+    exit 0
+fi
 
-(test "$isreject" != "1" && $gitcmd am -3 $file) || (
-	echo "Failures - Wiggling"
-	$gitcmd reset --hard
-	$gitcmd clean -f
-	errors=$($gitcmd apply --rej $file 2>&1)
-	echo "$errors" >> ~/patch.log
-	export missingfiles=""
-	export summaryfail=""
-	export summarygood=""
-	for i in $(find . -name \*.rej); do
-        	base=$(echo "$i" | sed 's/.rej//g')
-		if [ -f "$i" ]; then
-        		sed -e 's/^diff a\/\(.*\) b\/\(.*\)[[:space:]].*rejected.*$/--- \1\n+++ \2/' -i $i && wiggle -v -l --replace "$base" "$i"
-        		rm "$base.porig" "$i"
-		else
-			echo "No such file: $base"
-			missingfiles="$missingfiles\n$base"
-		fi
-	done
+if [ "$isreject" ] || [ "$($gitcmd am -3 "$file")" ]; then
+    echo "Failures - Wiggling"
+    $gitcmd reset --hard
+    $gitcmd clean -f
+    errors=$($gitcmd apply --rej "$file" 2>&1)
+    echo "$errors" >> patch.log # Anyone encountering this should just be told to fix it or knows already.
+    export missingfiles=""
+    export summaryfail=""
+    export summarygood=""
+    for i in $(find . -name '*.rej'); do
+        base="$(echo "$i" | sed 's/.rej//g')"
+        if [ -f "$i" ]; then
+            sed -e 's/^diff a\/\(.*\) b\/\(.*\)[[:space:]].*rejected.*$/--- \1\n+++ \2/' "$i" && wiggle -v -l --replace "$base" "$i"
+            rm -f "$base.porig" "$i"
+        else
+            echo "No such file: $base"
+            missingfiles="$missingfiles\n$base"
+        fi
+    done
 	for i in $($gitcmd status --porcelain | awk '{print $2}'); do
 		filedata=$(cat "$i")
-		if [ -f "$file" ] && [[ "$filedata" == *"<<<<<"* ]]; then
+		if [ -f "$file" ] && str_contains "$filedata" "<<<<<"; then
 			export summaryfail="$summaryfail\nFAILED TO APPLY: $i"
 		else
 			$gitcmd add "$i"
 			export summarygood="$summarygood\nAPPLIED CLEAN: $i"
 		fi
 	done
-	echo -e "$summarygood"
-	echo -e "$summaryfail"
-	if [[ "$errors" == *"No such file"* ]]; then
+	printf '%s' "$summarygood"
+	printf '%s' "$summaryfail"
```

Comment:
This won't interpret the newlines. Did you mean to use `'%b\n'` format string instead?
```suggestion
	printf '%b\n' "$summarygood"
	printf '%b\n' "$summaryfail"
```